docs/developer/style.md

```diff
@@ -2,34 +2,348 @@
 
 ## Table of Contents
 
-- [Introduction](#Introduction)
-- [Style](#Style)
-- [Test](#Test)
-	- [Table-Driven-Test](#Table-Driven-Test)
+- [Introduction](#introduction)
+- [Code Formatting and Naming Convension](#code-formatting-and-naming-convension)
+  - [Project Layout](#project-layout)
+  - [Packages](#packages)
+  - [Interfaces](#interfaces)
+  - [Structs](#structs)
+    - [Struct initialization](#struct-initialization)
+  - [Variables and Constant](#variables-and-constant)
+  - [Methods](#methods)
+    - [Getter and Setter](#getter-and-setter)
+  - [Error handling](#error-handling)
+  - [Logging](#logging)
+- [Program comments](#program-comments)
+- [Documentation](#documentation)
+- [Internal packages](#internal-packages)
+- [Dependency management and Build](#dependency-management-and-build)
+- [Test](#test)
+  - [Table-Driven-Test](#table-driven-test)
+  - [The steps to create a Table-Driven-Test](#the-steps-to-create-a-table-driven-test)
 
 ## Introduction
 
-## Style
+This guideline includes the coding style for all Vald contributors and reviewers. Everyone should follow this guideline to keep the style consistent so everyone can understand and contribute to Vald easier once they learn this guideline. You should have the basic knowledge of how to write Golang before contributing to Vald. If you found any bug please create [a GitHub issue](https://github.com/vdaas/vald/issues/new?assignees=&labels=type%2Fbug%2C+priority%2Fmedium%2C+team%2Fcore&template=bug_report.md&title=) and we will work on it.
+
+Please also read the [Contribution guideline](../../CONTRIBUTING.md) before you start contributing to Vald.
+
+## Code Formatting and Naming Convension
+
+Code formatting and naming conventions affect coding readability and maintainability. Every developer has a different coding style, luckily Golang provides tools to format source code and checking for the potential issue in the source code. We recommend using [goimports](https://github.com/golang/tools/tree/master/cmd/goimports) to format the source code in Vald, and [golangci-lint](https://github.com/golangci/golangci-lint) with `--enable-all` option. We suggest everyone install the plugin for your editor to format the code once you edit the code automatically.
```

Comment:
![badge](https://img.shields.io/badge/review-suggestion-blue.svg?style=flat-square)
it is recommended to use `make format` for running goimports. please add description about it.

Reply:
fixed, please review.

Reply:
Thanks! good job :+1: